### ISSUE TITLE
Modifying cookies should create a new response object

### ISSUE THREAD

**Issue by Catgroove** [honojs/hono#986]
Setting cookies via `c.cookie` does not work with `getMiniflareFetchMock` provided by Miniflare, this is because the headers object returned by `MockAgent` is immutable, and `c.cookie` mutates the headers instead of creating a new `Response`.

```
Trace: TypeError: immutable
    at Headers.append (/*/node_modules/.pnpm/undici@5.9.1/node_modules/undici/lib/fetch/headers.js:227:13)
    at Context.header (/*/node_modules/.pnpm/hono@3.0.5/node_modules/hono/dist/cjs/context.js:53:28)
    at Context.cookie (/*/node_modules/.pnpm/hono@3.0.5/node_modules/hono/dist/cjs/context.js:150:12)
```

**Comment by Catgroove** [comment]
Setting cookies does not seem to work in production regardless, so it's an issue with the `MockAgent`.

**Comment by Catgroove** [comment]
It seems to be middleware related. I have a middleware that sets a cookie based on some user state. I assume that the cookie method is intended to be used inside of handlers and not middlewares?

**Comment by Catgroove** [comment]
Update, this particular case was solved by the latest 3.1 release, but mocking using the `MockAgent` is still broken. I had to create my own fetch-mock.

**Comment by yusukebe** [comment]
Closing.
